databuilder/databuilder/task/search/search_metadata_to_elasticsearch_task.py

```diff
@@ -0,0 +1,161 @@
+            raise TypeError(msg)
+
+        self.elasticsearch_batch_size = conf.get(
+            SearchMetadatatoElasticasearchTask.ELASTICSEARCH_PUBLISHER_BATCH_SIZE, 10000
+        )
+        self.elasticsearch_timeout = conf.get(
+            SearchMetadatatoElasticasearchTask.ELASTICSEARCH_TIMEOUT, 120
+        )
+
+    def to_document(self, document_mapping: Document, metadata: Iterator, index: str) -> Document:
+        return document_mapping(_index=index, **metadata)
+
+    def generate_documents(self, record: Iterator) -> Generator:
+        # iterate through records
+        while record:
+            record = self.transformer.transform(record)
+            if not record:
+                # Move on if the transformer filtered the record out
+                record = self.extractor.extract()
+                continue
+            yield self.to_document(document_mapping=self.document_mapping,
+                                   metadata=record,
+                                   index=self.elasticsearch_new_index).to_dict(True)
+            record = self.extractor.extract()
+
+    def _delete_old_index(self, connection: Connections, document_index: Index) -> None:
+        alias_updates = [
+            {"add": {"index": document_index._name, "alias": self.elasticsearch_alias}}
+        ]
+        for index_name in connection.indices.get_alias():
+            if index_name.startswith(f"{self.elasticsearch_alias}_"):
+                if index_name != document_index._name:
+                    LOGGER.info(f"Deleting index old {index_name}")
+                    alias_updates.append({"remove_index": {"index": index_name}})
+        connection.indices.update_aliases({"actions": alias_updates})
+
+    def run(self) -> None:
+        LOGGER.info('Running search metadata to Elasticsearch task')
+        try:
+            # extract records from metadata store
+            record = self.extractor.extract()
+
+            # create connection
+            connections.add_connection('default', self.elasticsearch_client)
+            connection = connections.get_connection()
+
+            # health check ES
+            health = connection.cluster.health()
+            status = health["status"]
+            if status not in ("green", "yellow"):
+                msg = f"Elasticsearch healthcheck failed: {status}"
+                LOGGER.error(msg)
+                raise Exception(msg)
+
+            # create index
+            LOGGER.info(f"Creating ES index {self.elasticsearch_new_index}")
+            index = Index(name=self.elasticsearch_new_index, using=self.elasticsearch_client)
+            index.document(self.document_mapping)
+            index.create()
+
+            # publish search metadata to ES
+            cnt = 0
+            for success, info in parallel_bulk(connection,
+                                               self.generate_documents(record=record),
+                                               raise_on_error=False,
```

Comment:
does raise_on_error also imply raise_on_exception?

Reply:
https://github.com/elastic/elasticsearch-py/blob/5a6bfe6dd964c62d80626da5aa9e43fef7c9a29e/elasticsearch/helpers/actions.py#L306

It seems like, if you want to hide exceptions, you should set both. Otherwise, it may be better to keep both off so you get the "real" exception instead of the wrapped exception.